### PR TITLE
feat: add motionComponent prop to Reorder Group and Item components

### PR DESCRIPTION
**Pull request by james-wallis**
Resolves: https://github.com/framer/motion/issues/1930

I've tested this locally and it resolves the linked issue. I'm open to feedback/changes and can add tests/docs once the implementation details are good.

Edit: Changed this to add an `ignoreStrict` prop onto the `motion` component. When `ignoreStrict` is given, it will output a `warn` instead of throwing an error so that users are still aware of what using `motion` inside `LazyMotion` will do.

### CODE REVIEW

File: packages/framer-motion/src/components/Reorder/Item.tsx [hunk 1]
```diff
@@ -53,11 +61,12 @@ export function ReorderItem<V>(
         as = "li",
         onDrag,
         layout = true,
+        motionComponent = motion,
```
Comment (mattgperry) [inline packages/framer-motion/src/components/Reorder/Item.tsx:64]: This will fix the error message but the bundle size will still go up

**Comment by mattgperry** [comment]
Hi James, thanks for the PR!

The problem with this approach is it will remove the error message but because the prop can't be resolved until runtime, it will still increase the bundlesize in the way it's warning about.

Perhaps a cleaner approach would be to have a `ignoreStrict` option on `MotionConfig` or `motion` that could be used to disable the check, although we'd still have the same problem of an increased bundlesize. 

Alternatively the `invariant` could become a `warning` - on one hand I think it's reasonable for `Reorder` to expect the drag functionality to be present but on the other I think there's value in allowing it to still be lazy-loaded.

**Comment by james-wallis** [comment]
Hey @mattgperry, yeah, that makes sense!

I want to check my understanding of `LazyMotion` with you if that is alright. 
If I use `LazyMotion` and load asynchronously using `import()` and use `m` in all places in my app apart from on the `/reorder` page (using Next.js if that makes a difference) where `Reorder` is used - I'll still get the reduced bundle size when not on `/reorder`?
- I'm hoping this is the case so I'll work on a change to add the `ignoreStrict` prop to a `motion` component. Think it makes sense to prevent the error when using a built-in component?

Or if my app is going to `Reorder` I should use the `motion` component always?

**Comment by mattgperry** [comment]
@james-wallis Yeah this is the case - `Reorder` will come with `motion` and all the features. But this means when you're not on this page, the features won't be loaded.

**Review by mattgperry — APPROVED** [review]
Great PR! Thanks

**Pull request merged**